### PR TITLE
:seedling: adding arm images for windows and windows

### PR DESCRIPTION
**Pull request by shawn-hurley**
* Added ARM binary builds for GitHub actions
* Added back specific log file for conveyor-analyzer-rpc server. This is because the logs were getting to hard to debug actual issues in both `kai_analyzer_rpc` and `kai_rpc_server`. 
* Added back the ability to trace log.
* Carry Jonah's patches for fixing #512 
* Added fix in `kai_analyzer_rpc` to allow for gracefully shutting down jdtls on completion. 
* Updated analyzer-lsp to latest on main to get the `fmt.Printf` fix from @pranavgaikwad and the gracefull shutdown of the java provider.

### CODE REVIEW

**Comment by shawn-hurley** [comment]
NOTE: that the successful run was not actually successful

File: kai/reactive_codeplanner/task_manager/task_manager.py [hunk 1]
```diff
@@ -213,14 +213,14 @@ def handle_new_tasks_after_processing(self, task: Task) -> None:
...
-            ]
-        )
+        similar_non_resolved_tasks = set()
+
+        for t in resolved_tasks:
+            logger.info("is task similar to unprocessed tasks")
+            for u in unprocessed_new_tasks:
+                if self.is_similar_to_task(t, u):
+                    logger.debug("adding task to similar_non_resolved_task: %s", t)
+                    similar_non_resolved_tasks.add(t)
```
Comment (JonahSussman) [inline kai/reactive_codeplanner/task_manager/task_manager.py:223]: Do we want to keep these log messages?

File: kai/reactive_codeplanner/task_runner/analyzer_lsp/api.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ def __str__(self) -> str:
         else:
             shadowed_priority = self.__class__.priority
 
-        return f"{self.__class__.__name__}<loc={self.file}:{self.line}:{self.column}, message={self.violation.description}>(priority={self.priority}({shadowed_priority}), depth={self.depth}, retries={self.retry_count})"
+        return f"{self.__class__.__name__}<loc={self.file}:{self.line}:{self.column}, message={self.violation.id}>(priority={self.priority}({shadowed_priority}), depth={self.depth}, retries={self.retry_count})"
```
Comment (JonahSussman) [inline kai/reactive_codeplanner/task_runner/analyzer_lsp/api.py:30]: Let's change `message=` to something like `violation.id=`

File: .github/workflows/build-and-push-binaries.yml [hunk 3]
```diff
@@ -104,7 +121,7 @@ jobs:
           git config --global user.email "no-reply@konveyor.io"
           git config --global user.name "Konveyor CI"
           git clone https://github.com/konveyor/rulesets
```
Comment (JonahSussman) [inline .github/workflows/build-and-push-binaries.yml:123]: Do we want to use the `make` targets for the CI?

**Review by JonahSussman — APPROVED** [review]
Looks good, a few things

File: .github/workflows/build-and-push-binaries.yml (hunk 3 above)
Comment (shawn-hurley) [reply .github/workflows/build-and-push-binaries.yml:123]: yes, but also followup?

File: kai/reactive_codeplanner/task_runner/analyzer_lsp/api.py (hunk 2 above)
Comment (shawn-hurley) [reply kai/reactive_codeplanner/task_runner/analyzer_lsp/api.py]: fixed

File: .github/workflows/build-and-push-binaries.yml (hunk 3 above)
Comment (JonahSussman) [reply .github/workflows/build-and-push-binaries.yml:123]: Sure thing, I'll make an issue.

Comment (JonahSussman) [reply .github/workflows/build-and-push-binaries.yml:123]: #520

**Review by pranavgaikwad — APPROVED** [review]

**Pull request merged**